packages/fetchai/skills/aries_alice/handlers.py

```diff
@@ -0,0 +1,236 @@
+
+"""This package contains the handlers for the aries_alice skill."""
+
+import base64
+import binascii
+import json
+from typing import Dict, Optional, cast
+from urllib.parse import urlparse
+
+from aea.configurations.base import ProtocolId, PublicId
+from aea.mail.base import Envelope, EnvelopeContext
+from aea.protocols.base import Message
+from aea.protocols.default.message import DefaultMessage
+from aea.skills.base import Handler
+
+from packages.fetchai.protocols.http.message import HttpMessage
+from packages.fetchai.protocols.http.serialization import HttpSerializer
+
+HTTP_CONNECTION_PUBLIC_ID = PublicId("fetchai", "http_client", "0.1.0")
+
+HTTP_PROTOCOL_PUBLIC_ID = PublicId("fetchai", "http", "0.1.0")
+DEFAULT_ADMIN_HOST = "127.0.0.1"
+DEFAULT_ADMIN_PORT = 8031
+
+
+class AriesDemoDefaultHandler(Handler):
+    """This class represents alice's handler for default messages."""
+
+    SUPPORTED_PROTOCOL = DefaultMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def __init__(self, **kwargs):
+        """Initialize the handler."""
+        self.admin_host = kwargs.pop("admin_host", DEFAULT_ADMIN_HOST)
+        self.admin_port = kwargs.pop("admin_port", DEFAULT_ADMIN_PORT)
+
+        super().__init__(**kwargs)
+
+        self.admin_url = "http://{}:{}".format(self.admin_host, self.admin_port)
+
+        self.kwargs = kwargs
+        self.handled_message = None
+
+    def _admin_post(self, path: str, content: Dict = None):
+        # Request message & envelope
+        request_http_message = HttpMessage(
+            performative=HttpMessage.Performative.REQUEST,
+            method="POST",
+            url=self.admin_url + path,
+            headers="",
+            version="",
+            bodyy=b"" if content is None else json.dumps(content).encode("utf-8"),
+        )
+        context = EnvelopeContext(connection_id=HTTP_CONNECTION_PUBLIC_ID)
+        envelope = Envelope(
+            to="Alice_ACA",
+            sender=self.context.agent_address,
+            protocol_id=HTTP_PROTOCOL_PUBLIC_ID,
+            context=context,
+            message=HttpSerializer().encode(request_http_message),
+        )
+        self.context.outbox.put(envelope)
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to an envelope.
+
+        :param message: the message
+        :return: None
+        """
+        message = cast(DefaultMessage, message)
+        if message.performative == DefaultMessage.Performative.BYTES:
+            content_bytes = message.content
+            content = json.loads(content_bytes)
+            self.context.logger.info("Received message content:" + str(content))
+            if "@type" in content:
+                # self.handle_received_invite(content)
+                self._admin_post("/connections/receive-invitation", content)
```

Comment:
so here you are hardcoding a path. It's not clear why, but in any case, hardcoded values should be declared as a constant on the top of the file.

Reply:
Done